packages/measure/src/measure-async-function.tsx

```diff
@@ -0,0 +1,42 @@
+import { config } from './config';
+import type { MeasureResults } from './types';
+import { type RunResult, getCurrentTime, processRunResults } from './measure-helpers';
+import { showFlagsOutputIfNeeded, writeTestStats } from './output';
+import { MeasureFunctionOptions } from './measure-function';
+
+export interface MeasureAsyncFunctionOptions extends MeasureFunctionOptions {}
+
+export async function measureAsyncFunction(
+  fn: () => Promise<any>,
+  options?: MeasureAsyncFunctionOptions
+): Promise<MeasureResults> {
+  const stats = await measureAsyncFunctionInternal(fn, options);
+
+  if (options?.writeFile !== false) {
+    await writeTestStats(stats, 'function');
```

Comment:
Extend the `MeasureType` and related stuff like `type-schemas.ts`
```suggestion
    await writeTestStats(stats, 'async-function');
```